garage/tf/samplers/off_policy_vectorized_sampler.py

```diff
@@ -0,0 +1,155 @@
+                input_obses = np.concatenate((obs, d_g), axis=-1)
+            else:
+                input_obses = obses
+            if self.algo.es:
+                actions, agent_infos = self.algo.es.get_actions(
+                    input_obses, self.algo.policy)
+            else:
+                actions, agent_infos = self.algo.policy.get_actions(
+                    input_obses)
+
+            next_obses, rewards, dones, env_infos = self.vec_env.step(actions)
+            agent_infos = tensor_utils.split_tensor_dict_list(agent_infos)
+            env_infos = tensor_utils.split_tensor_dict_list(env_infos)
+            if agent_infos is None:
+                agent_infos = [dict() for _ in range(self.vec_env.num_envs)]
+            if env_infos is None:
+                env_infos = [dict() for _ in range(self.vec_env.num_envs)]
+
+            if self.algo.replay_buffer_type == Buffer.REGULAR:
+                self.algo.replay_buffer.add_transition(
+                    observation=obses,
+                    action=actions,
+                    reward=rewards * self.algo.reward_scale,
+                    terminal=dones,
+                    next_observation=next_obses,
+                )
+            elif self.algo.replay_buffer_type == Buffer.HER:
+                info_dict = {
+                    "info_{}".format(key): []
+                    for key in env_infos[0].keys()
+                }
+                for env_info in env_infos:
+                    for key in env_info.keys():
+                        info_dict["info_{}".format(key)].append(
+                            env_info[key].reshape(1))
+                self.algo.replay_buffer.add_transition(
+                    observation=obs,
+                    action=actions,
+                    goal=d_g,
+                    achieved_goal=a_g,
+                    terminal=dones,
+                    **info_dict,
+                )
+
+            if self.algo.replay_buffer_type == Buffer.HER and (
+                    rollout == self.algo.max_path_length - 1):
+                self.algo.replay_buffer.add_transition(
+                    observation=[
+                        next_obs['observation'] for next_obs in next_obses
+                    ],
+                    achieved_goal=[
+                        next_obs['achieved_goal'] for next_obs in next_obses
+                    ])
+
+            for idx, reward, env_info, done in zip(itertools.count(), rewards,
+                                                   env_infos, dones):
+                if running_paths[idx] is None:
+                    running_paths[idx] = dict(
+                        rewards=[],
+                        env_infos=[],
+                    )
+                running_paths[idx]["rewards"].append(reward)
+                running_paths[idx]["env_infos"].append(env_info)
+
+                if self.algo.replay_buffer_type == Buffer.REGULAR:
+                    add_to_paths = done or (
```

Comment:
Maybe I don't understand the method, but why don't we add to the paths when done if using HER?
I might just be confused by the bigger picture somehow, but it looks like this logic can be simplified?